cli/create.go

```diff
@@ -100,6 +101,25 @@ func create(containerID, bundlePath, console, pidFilePath string, detach bool,
 		return err
 	}
 
+	if runtimeConfig.FactoryConfig.Template {
+		factoryConfig := vf.Config{
+			Template: true,
+			VMConfig: vc.VMConfig{
+				HypervisorType:   runtimeConfig.HypervisorType,
+				HypervisorConfig: runtimeConfig.HypervisorConfig,
+				AgentType:        runtimeConfig.AgentType,
+				AgentConfig:      runtimeConfig.AgentConfig,
+			},
+		}
+		kataLog.WithField("factory", factoryConfig).Info("load vm factory")
+		f, err := vf.NewFactory(factoryConfig, true)
+		if err != nil {
+			kataLog.WithError(err).Info("load vm factory failed")
```

Comment:
I think it's worth adding a comment here explaining that if vm factory support fails, the system will "degrade" automatically to using the non-factory codepath.

In fact, I think we might want to add another variable to the config file to avoid this graceful degradation as some users only want to use VM factories:

```
[factory]

# If set, fail if VM factory support is not available.
## strict=true
```

Reply:
Yes, agreed. But again, I'd like to push all these UX changes to later PRs. This PR is mostly to add core vm factory functionalities.